release/nightly_tests/dataset/iter_batches_benchmark.py

```diff
@@ -0,0 +1,99 @@
+import sys
+from typing import Optional
+
+import ray
+from ray.data.dataset import Dataset
+
+from benchmark import Benchmark
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+def iter_batches(
+    ds: Dataset,
+    batch_size: Optional[int] = None,
+    batch_format: Literal["default", "pandas", "pyarrow", "numpy"] = "default",
+    local_shuffle_buffer_size: Optional[int] = None,
+    use_default_params: bool = False,
+) -> Dataset:
+    num_batches = 0
+    if use_default_params:
+        for batch in ds.iter_batches():
+            num_batches += 1
+    else:
+        for batch in ds.iter_batches(
+            batch_format=batch_format,
+            batch_size=batch_size,
+            local_shuffle_buffer_size=local_shuffle_buffer_size,
+        ):
+            num_batches += 1
+    print("iter_batches done, num_rows:", ds.count(), "num_batches:", num_batches)
+    return ds
+
+
+def run_iter_batches_benchmark(benchmark: Benchmark):
+    ds = (
+        ray.data.read_parquet(
+            "s3://anonymous@air-example-data/ursa-labs-taxi-data/by_year/2018/01"
```

Comment:
How many records are in this file, i.e. how many records are there in each of the 24 blocks? A quick poke suggest 8759874 records, so ~364995 records per block.

It might be good to comment the number of records in the file in the code, and/or log out the number of records in the benchmark.

Reply:
Added.